File: Week 06/Problem Set 06/sentimental-readability/readability.py
from cs50 import get_string
s = get_string("Text: ")

L, S = 0, 0
for c in s:
    if c.isalpha():
        L += 1
    if c == "." or c == "!" or c == "?":
        S += 1

G = round(0.0588 * 100 * L / len(s.split()) - 0.296 * 100 * S / len(s.split()) - 15.8)
if G < 1:
    print("Before Grade 1")
elif G >= 16:
    print("Grade 16+")
else:
    print(f"Grade {G}")

File: Week 02/Problem Set 02/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main (void) {
    string text = get_string("Text: ");
    int W = 1, L = 0, S = 0;

    for (int i = 0; i < strlen(text); i++) if (text[i] == ' ') W++;
    for (int i = 0; i < strlen(text); i++) if (toupper(text[i]) >= 65 && toupper(text[i]) <= 90) L++;
    for (int i = 0; i < strlen(text); i++) if (text[i] == '.' || text[i] == '!' || text[i] == '?') S++;

    int index = round(0.0588 * 100 * ((float) L / (float) W) - 0.296 * 100 * ((float) S / (float) W) - 15.8);
    if (index >= 16) printf("Grade 16+\n");
    else if (index < 1) printf("Before Grade 1\n");
    else printf("Grade %i\n", index);
}